module/VuFind/src/VuFind/Auth/OpenIDConnect.php

```diff
@@ -360,6 +364,50 @@ public function getSessionInitiator(string $target): ?string
         return $this->getProvider()->authorization_endpoint . '?' . http_build_query($params);
     }
 
+    /**
+     * Perform cleanup at logout time.
+     *
+     * @param string $url URL to redirect user to after logging out.
+     *
+     * @return string Redirect URL (modified for OpenIDConnect logout).
+     */
+    public function logout($url)
+    {
+        $redirectUrl = $url;
+        $end_session_endpoint = false;
+
+        $logout = $this->getConfig('logout');
+
+        if (!$logout) {
+            // No logout configured, so don't logout from service provider.
+            $this->debug('no logout URL given');
+        } elseif (filter_var($logout, FILTER_VALIDATE_URL)) {
+            // A valid URL was configured, use it.
+            $end_session_endpoint = $logout;
+        } else {
+            // Get end_session_endpoint from provider.
+            $provider = $this->getProvider();
+            $end_session_endpoint = $provider->end_session_endpoint;
```

Comment:
Should we allow for a missing value to avoid notices?

```suggestion
            $end_session_endpoint = $provider->end_session_endpoint ?? null;
```

Reply:
Good idea. I don't know whether existing providers can trigger this case, but in any case it makes the code more robust.